src/test/shell/testenv.sh

```diff
@@ -654,6 +654,11 @@ function cleanup_workspace() {
 }
 
 function testenv_tear_down() {
+  if [ -d "${WORKSPACE_DIR:-}" ]; then
+    # Try to shutdown Bazel at the end to prevent log spam.
+    cd "${WORKSPACE_DIR}"
+    try_with_timeout bazel shutdown || true
```

Comment:
Note that @philwo carefully removed a lot of `bazel shutdown`s a while ago to speed up our tests by "a lot". This would revert this speedup.

Reply:
Yes. We can’t do this. This would completely kill our test performance.

Reply:
I wasn't proposing it seriously, I assumed there are performance issues. Is there another way we could address #12397?

Reply:
Perhaps @meteorcloudy has an idea?

Reply:
You cannot delete a directory on Windows until its file handle is released by other processes. `try_with_timeout` was [introduced](https://cs.opensource.google/bazel/bazel/+/master:src/test/shell/testenv.sh;l=246?q=try_with_timeout&ss=bazel) to solve this problem. In #12397, the error message was generated by `try_with_timeout rm -fr target_skipping`, to reduce the noise, maybe changing `try_with_timeout` to redirect the error message to `/dev/null`. See https://cs.opensource.google/bazel/bazel/+/master:src/test/shell/testenv.sh;l=248?q=try_with_timeout&ss=bazel